src/qt/sendcoinsdialog.cpp

```diff
@@ -436,14 +442,76 @@ void SendCoinsDialog::on_sendButton_clicked()
         accept();
         CoinControlDialog::coinControl()->UnSelectAll();
         coinControlUpdateLabels();
-        uint256 hashCurrentTx = currentTransaction.getWtx()->get().GetHash();
+        uint256 hashCurrentTx = prepareData->tx->getWtx()->get().GetHash();
         if (fDandelion) {
             LOCK(veil::dandelion.cs);
             veil::dandelion.Add(hashCurrentTx, GetAdjustedTime() + veil::dandelion.nDefaultStemTime, veil::dandelion.nDefaultNodeID);
         }
         Q_EMIT coinsSent(hashCurrentTx);
     }
     fNewRecipientAllowed = true;
+    SetTransactionLabelState(TxPrepState::DONE);
+}
+
+void SendCoinsDialog::on_sendButton_clicked()
+{
+    if(!model || !model->getOptionsModel())
+        return;
+
+    QList<SendCoinsRecipient> recipients;
+    bool valid = true;
+
+    std::string error;
+    for(int i = 0; i < ui->entries->count(); ++i)
+    {
+        SendCoinsEntry *entry = qobject_cast<SendCoinsEntry*>(ui->entries->itemAt(i)->widget());
+        if(entry) {
+            if(entry->validate(model->node(), error)) {
+                recipients.append(entry->getValue());
+            }
+            else {
+                valid = false;
+            }
+        }
+    }
+
+    if(!valid){
+        openToastDialog(QString::fromStdString("Invalid data\n" + error), this);
+        return;
+    }
+
+    if(recipients.isEmpty()){
+        openToastDialog("No recipients", this);
+        return;
+    }
+
+    fNewRecipientAllowed = false;
+    WalletModel::UnlockContext ctx(model->requestUnlock());
+    if(!ctx.isValid())
+    {
+        // Unlock wallet was cancelled
+        fNewRecipientAllowed = true;
+        return;
+    }
+
+    prepareData = new PrepareTxData();
```

Comment:
This is probably a memleak here. Should change `prepareData` to be a smart pointer for automatic deletion.

Reply:
Agreed. Every time the send button is clicked new memory is being allocated for the pointer.